extensions/ql-vscode/src/common/short-paths.ts

```diff
@@ -0,0 +1,111 @@
+import { platform } from "os";
+import { basename, dirname, join, normalize, resolve } from "path";
+import { lstat, readdir } from "fs/promises";
+import { extLogger } from "./logging/vscode";
+
+async function log(message: string): Promise<void> {
+  await extLogger.log(message);
```

Comment:
If we're not removing it, I'd suggest using the `BaseLogger` interface instead of using `extLogger` directly. All of the code in this file is independent of VS Code and could be tested using unit tests, but adding the `extLogger` import adds a dependency on the `vscode` module, which requires any tests to be located in the `vscode-tests` directory.

Reply:
This code is only executed on machines with a particular legacy feature enabled, and even then only if we actually see an 8.3 filename. Outside of our CI machines, which have this enabled, I expect that fewer than 1% of customers will ever get a single line logged from this code. On the other hand, if something does go wrong in this code, we're unlikely to notice without a log message, and it will be difficult to reproduce locally. So, I'd prefer to leave them in.

I did convert everything over to use `BaseLogger`, though, which solved one of Robert's comments and will make it easier to write a unit test in the future if we decide to.